dynawo/sources/Common/Dictionaries/validateDictionaries.py

```diff
@@ -12,161 +12,224 @@
 # simulation tool for power systems.
 
 ##
-# Small python utility to create header and CPP files associated to a dictionary
+# Small python utility to check if dictionaries have the same entries and message description.
+# It optionally creates header, CPP and Modelica files associated to each dictionary.
 #
 
-import glob
-import formatter
-import string
 import filecmp
-import shutil
+import glob
 import os
+import shutil
+
 from optparse import OptionParser
 
+
 ##
-# Class containing all dictionaries found by the utility
-class Dictionaries:
+# Class containing all dictionaries found by the utility.
+class DictionariesPool:
     ##
-    # Default constructor
-    def __init__( self ):
+    # Default constructor.
+    #
+    # @param  self: the object pointer
+    def __init__(self):
         ## dictionaries found
-        self.dicts_ = []
-        ## name of each dictionary
-        self.names_ = []
+        self._dictionaries = {}
 
     ##
-    # Add a new dictionary to the container
-    # @param self: object pointer
-    # @param dictionary : new dictionary to add
+    # Add a new dictionary to the container.
+    #
+    # @param  self: the object pointer
+    # @param  dictionary: the dictionary to add
     # @return
-    def add_dict(self, dictionary):
-        index = [x for x in self.names_ if x == dictionary.name()]
-        if index:
-            self.dicts_[self.names_.index(dictionary.name())].dict_.update(dictionary.dict_)
+    def add_dictionary(self, dictionary):
+        full_name = dictionary.full_name
+        if full_name in self._dictionaries:
+            self._dictionaries[full_name].extend(dictionary)
```

Comment:
Why not adding a structure inside Dictionary class to store all existing languages?

Reply:
A Dictionary is a bunch of messages parsed from several files for one language.
A DictionariesPool is a pool of Dictionary.
Why not make language discrimination in this last class with an internal dict (key will be name without language) of list (or dict with language as key) of Dictionary instead of the actual dict (with full name as key) of Dictionary ?

Reply:
Internal dict in DictionariesPool keeps now track of locale.